components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java

```diff
@@ -952,4 +890,65 @@ private OpenAPI getOpenAPI(String oasDefinition) throws APIManagementException {
         }
         return parseAttemptForV3.getOpenAPI();
     }
+
+    /**
+     * Construct openAPI definition for graphQL. Add get and post operations
+     *
+     * @param openAPI OpenAPI
+     * @return modified openAPI for GraphQL
+     */
+    private void modifyGraphQLSwagger(OpenAPI openAPI) {
+        SwaggerData.Resource resource = new SwaggerData.Resource();
+        resource.setAuthType(APIConstants.AUTH_APPLICATION_OR_USER_LEVEL_TOKEN);
+        resource.setPolicy(APIConstants.DEFAULT_SUB_POLICY_UNLIMITED);
+        resource.setPath("/*");
+
+        resource.setVerb(APIConstants.HTTP_GET);
+        Operation getOperation = createOperation(resource);
+        resource.setVerb(APIConstants.HTTP_POST);
+        Operation postOperation = createOperation(resource);
+
+        //get operation
+        Parameter getParameter = new Parameter();
+        getParameter.setName(APIConstants.GRAPHQL_QUERY);
+        getParameter.setIn("query");
+        getParameter.setRequired(true);
+        getParameter.setDescription("Query to be passed to graphQL API");
+
+        Schema getSchema = new Schema();
+        getSchema.setType("string");
+        getParameter.setSchema(getSchema);
+        getOperation.addParametersItem(getParameter);
+
+        //post operation
+        RequestBody requestBody = new RequestBody();
+        requestBody.setDescription("Query or mutation to be passed to graphQL API");
+        requestBody.setRequired(true);
+
+        JSONObject typeOfPayload = new JSONObject();
+        JSONObject payload = new JSONObject();
+        typeOfPayload.put(APIConstants.TYPE, "string");
```

Comment:
Lets use constant string

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/8244